cluster_management/src/main/java/com/pinterest/rocksplicator/HelixClient.java

```diff
@@ -0,0 +1,164 @@
+import org.apache.commons.cli.OptionBuilder;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+import org.apache.helix.HelixAdmin;
+import org.apache.helix.HelixManager;
+import org.apache.helix.manager.zk.ZKHelixAdmin;
+import org.apache.helix.model.ExternalView;
+import org.apache.log4j.BasicConfigurator;
+import org.apache.log4j.ConsoleAppender;
+import org.apache.log4j.Level;
+import org.apache.log4j.PatternLayout;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class HelixClient {
+
+  private static final Logger LOG = LoggerFactory.getLogger(Participant.class);
+  private static final String zkServer = "zkSvr";
+  private static final String cluster = "cluster";
+  private static final String resource = "resource";
+  private static final String partition = "partition";
+
+  private static HelixManager helixManager;
+
+  private static Options constructCommandLineOptions() {
+    Option zkServerOption =
+        OptionBuilder.withLongOpt(zkServer).withDescription("Provide zookeeper addresses").create();
+    zkServerOption.setArgs(1);
+    zkServerOption.setRequired(true);
+    zkServerOption.setArgName("ZookeeperServerAddresses(Required)");
+
+    Option clusterOption =
+        OptionBuilder.withLongOpt(cluster).withDescription("Provide cluster name").create();
+    clusterOption.setArgs(1);
+    clusterOption.setRequired(true);
+    clusterOption.setArgName("Cluster name (Required)");
+
+    Option resourceOption =
+        OptionBuilder.withLongOpt(resource).withDescription("Provide resource name").create();
+    resourceOption.setArgs(1);
+    resourceOption.setRequired(true);
+    resourceOption.setArgName("Resource name (Required)");
+
+    Option partitionOption =
+        OptionBuilder.withLongOpt(partition).withDescription("Provide partition name").create();
+    partitionOption.setArgs(1);
+    partitionOption.setRequired(true);
+    partitionOption.setArgName("Partition name (Required)");
+
+    Options options = new Options();
+    options.addOption(zkServerOption)
+           .addOption(clusterOption)
+           .addOption(resourceOption)
+           .addOption(partitionOption);
+    return options;
+  }
+
+  private static CommandLine processCommandLineArgs(String[] cliArgs) throws ParseException {
+    CommandLineParser cliParser = new GnuParser();
+    Options cliOptions = constructCommandLineOptions();
+    return cliParser.parse(cliOptions, cliArgs);
+  }
+
+  /**
+   * Given the names of a cluster, resource and partition, this function returns the IP of the
+   * instance which hosts the leader replica of the partition. Returns an empty string in case
+   * an error or if there is no leader. 
+   */
+  public static String getLeaderHostIP(String zkConnectString,
+                                       String clusterName,
+                                       String resourceName,
+                                       String partitionName) {
```

Comment:
Can we make sure rocksplicator-replicator code doesn't need to know about the partitionNaming format helix uses? Is is possible?

Reply:
In the replicator code we have only the DB name (which has different formatting compared to the partition name). So once I get to the replicator code I'll figure out the best way to handle this.

Reply:
Thanks.